Source/Tools/LinuxEmulation/LinuxSyscalls/FileManagement.cpp

```diff
@@ -515,23 +615,63 @@ static bool ShouldSkipOpenInEmu(int flags) {
   return false;
 }
 
+bool FileManager::ReplaceEmuFd(int fd, int flags, uint32_t mode) {
+  char Tmp[PATH_MAX + 1];
+
+  if (fd < 0) {
+    return false;
+  }
+
+  // Get the path of the file we just opened
+  auto PathLength = FEX::get_fdpath(fd, Tmp);
+  if (PathLength == -1) {
+    return false;
+  }
+  Tmp[PathLength] = '\0';
+
+  // And try to open via EmuFD
+  auto EmuFd = EmuFD.Open(Tmp, flags, mode);
+  if (EmuFd == -1) {
+    return false;
+  }
+
+  // If we succeeded, swap out the fd
+  ::dup2(EmuFd, fd);
```

Comment:
Is there a reason to do this? We open a tmpfd for every emulatedfd. afaik there shouldn't be a need to dup the fd returned.

Reply:
Oh no, I see what you're doing here. You're duplicated over the previously opened FD. never mind.